src/libultra/io/viswapcontext.c

```diff
@@ -1,3 +1,66 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/viswapcontext/__osViSwapContext.s")
+void __osViSwapContext(void) {
+    register OSViMode* viMode;
+    register __OSViContext* viNext;
+    u32 origin;
+    u32 hStart;
+    u32 vstart;
+    u32 sp34;
+    u32 field;
+    register u32 s2;
```

Comment:
```suggestion
    uintptr_t origin;
    uintptr_t hStart;
    uintptr_t vstart;
    u32 sp34;
    u32 field;
    register uintptr_t s2;
```

Reply:
Are the top 3 actually pointers? They certainly don't look like it in the struct that this refers to: for example, from 
```c
OSViMode osViModePalLan1 = {
[...]
        0x800300,     // hStart
[...]
          0x280,    // origin
[...]
          0x5F0239, // vStart
      } },
};
```
they just seem to be magic numbers.

Reply:
They look like pointers just in the context of this file. Maybe someone with more knowledge of libultra can chime in, like @Thar0

Reply:
We discussed this a while ago in Discord. More research showed that `vstart` and `hstart` were not pointers, but that the use of `origin` was inconsistent, but since it's set in a hardware register, I think there ended up being agreement it should stay as `u32`? `s2` technically should be, but it's also part of a hardware register, so I'm not convinced either way, really. Again, @Thar0 may have a better idea of how we should handle this to be consistent.